### PR TITLE
fix(confluence-connector): clean up orphaned files and scopes when spaces are removed

### PR DESCRIPTION
**Pull request by lorand93**
## Summary
- Detect removed Confluence spaces by comparing existing scope children against discovered spaceKeys after each sync cycle
- Delete orphaned files via `files.deleteByKeyPrefix` and scopes via `scopes.delete` for spaces no longer in discovery results
- Enrich scope externalId to include Confluence spaceId (`confc:{tenant}:{spaceId}:{spaceKey}`) so partialKey can be reconstructed for removed spaces
- Per-space error handling ensures one cleanup failure does not block others

## Sync flow with cleanup

Shows where the new cleanup step fits into the existing sync cycle.

```mermaid
sequenceDiagram
    participant Sync as ConfluenceSyncService
    participant Scanner as ConfluencePageScanner
    participant Diff as FileDiffService
    participant Scope as ScopeManagementService
    participant Ingest as IngestionService
    participant API as Unique API

    Sync->>Scanner: discoverPages()
    Scanner-->>Sync: pages + attachments

    Note over Sync: Build spaceKeyToSpaceId map<br/>from ALL discovered items

    Sync->>Diff: computeDiff(pages, attachments)
    Diff-->>Sync: new / updated / deleted items

    opt Items to process
        Sync->>Scope: ensureSpaceScopes(spaceKeys, spaceKeyToSpaceId)
        Note over Scope: Migrates old externalId format<br/>to new format if needed
        Scope-->>Sync: spaceKey → scopeId map
        Sync->>Ingest: fetchAndIngestPages / ingestAttachments
    end

    rect rgb(255, 235, 200)
        Note over Sync,API: New: Cleanup phase
        Sync->>Scope: cleanupRemovedSpaces(discoveredSpaceKeys)
        Scope->>API: scopes.listChildren(rootScopeId)
        API-->>Scope: existing child scopes

        loop Each orphaned scope
            Scope->>API: files.deleteByKeyPrefix(partialKey)
            Scope->>API: scopes.delete(scopeId)
        end
    end
```

## Cleanup decision flow

How `cleanupRemovedSpaces` identifies and handles orphaned scopes.

```mermaid
flowchart TD
    A[cleanupRemovedSpaces called] --> B{discoveredSpaceKeys<br/>is empty?}
    B -- Yes --> C[Warn and skip<br/>Possible API issue]
    B -- No --> D[List children of root scope]
    D --> E[For each child scope]
    E --> F{Has parseable<br/>externalId?}
    F -- No --> G[Warn: missing or old format<br/>Skip this scope]
    F -- Yes --> H{spaceKey in<br/>discoveredSpaceKeys?}
    H -- Yes --> I[Still active, skip]
    H -- No --> J[Orphaned]
    J --> K[deleteByKeyPrefix partialKey]
    K --> L[delete scope]
    L --> M{Error?}
    M -- No --> N[Log success]
    M -- Yes --> O[Log error<br/>Continue with next scope]
```

## ExternalId format change

```mermaid
flowchart LR
    subgraph Before
        OLD["confc:{tenant}:{spaceKey}"]
    end

    subgraph After
        NEW["confc:{tenant}:{spaceId}:{spaceKey}"]
    end

    OLD -- "ensureSpaceScopes<br/>migrates on next<br/>content change" --> NEW

    NEW --> PARSE["parseExternalId()"]
    PARSE --> KEY["buildPartialKey()<br/>{spaceId}_{spaceKey}"]
    KEY --> DEL["files.deleteByKeyPrefix()"]
```

## Test plan
- [x] Unit tests for `parseExternalId` / `buildExternalId` (8 tests)
- [x] Behavioral tests for `cleanupRemovedSpaces` orphan detection and deletion flow (8 tests)
- [x] Integration tests for `synchronize()` calling cleanup with correct spaceKeys (2 tests)
- [x] Edge cases: missing/old externalId, per-space error isolation, V1/V2 key format
- [x] All 276 existing tests pass
- [x] biome check clean, tsc --noEmit clean

Closes UN-18296

### CODE REVIEW

**Comment by Ostrzy** [comment]
Claude Code didn't find anything worth mentioning apart from this

>   Scopes created before this branch have externalId format `confc:tenant:spaceKey` (2 segments). `parseExternalId` requires 3 segments (`confc:tenant:spaceId:spaceKey`) and    
  returns `undefined` for old ones — causing `identifyOrphanedScopes` to skip them with a warning.
                                                                                                                                                                                
 > The migration in `ensureSpaceScopes` only runs for **discovered** spaces. An orphaned space is, by definition, no longer discovered — so its externalId will never be migrated
   and it will be skipped forever.

Do we care much about migration or on existing deployment you will simply nuke everything and re-sync? 🤔

File: services/confluence-connector/src/utils/key-format.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+  if (segments.length !== 3) {
+    return undefined;
+  }
+
+  const [tenantName, spaceId, spaceKey] = segments as [string, string, string];
+  return { tenantName, spaceId, spaceKey };
+}
+
+export function buildExternalId(tenantName: string, spaceId: string, spaceKey: string): string {
```
Comment (Ostrzy) [inline services/confluence-connector/src/utils/key-format.ts]: Maybe worth mentioning that it's `buildScopeExternalId` - I know that we know that only scopes have this, but it's not obvious 🙂

**Review by Ostrzy — APPROVED** [review]

**Comment by lorand93** [comment]
> Do we care much about migration or on existing deployment you will simply nuke everything and re-sync? 🤔

Yes, I also discussed this matter with opus and even added an if clause to update the external id but it just introduced extra logic that becomes legacy after deploying in our QA environment. So instead of keeping that I can just nuke what we have in QA.

File: services/confluence-connector/src/utils/key-format.ts (hunk 1 above)
Comment (lorand93) [reply services/confluence-connector/src/utils/key-format.ts]: good point, I had troubles myself identifying which is which at one point
